src/fiatExchanges/normalizeQuotes.ts

```diff
@@ -0,0 +1,194 @@
+  return [
+    ...normalizeFiatConnectQuotes(flow, fiatConnectQuotes),
+    ...normalizeExternalProviders(flow, externalProviders),
+  ].sort(sortQuotesByFee)
+}
+
+export const sortQuotesByFee = (
+  { quote: quote1 }: NormalizedQuote,
+  { quote: quote2 }: NormalizedQuote
+) => {
+  const providerFee1 = quote1.fee ?? 0
+  const providerFee2 = quote2.fee ?? 0
+
+  return providerFee1 > providerFee2 ? 1 : -1
+}
+
+export function normalizeFiatConnectQuotes(
+  flow: CICOFlow,
+  quotes: FiatConnectQuote[] | undefined
+): NormalizedQuote[] {
+  const normalizedQuotes: NormalizedQuote[] = []
+
+  quotes?.forEach((quote) => {
+    // Iterate through every FiatAccountType. A single quote can have multiple
+    Object.entries(quote.fiatAccount).forEach(
+      ([key, value]: [string, FiatAccountTypeQuoteData | undefined]) => {
+        // Check if we support the FiatAccountType and at least one of the FiatAccountSchemas
+        // We choose the first FiatAccountSchema that is supported
+        const schema = value?.fiatAccountSchemas.find((schema) =>
+          SUPPORTED_FIAT_ACCOUNT_SCHEMAS.has(schema.fiatAccountSchema)
+        )
+        if (SUPPORTED_FIAT_ACCOUNT_TYPES.has(key as FiatAccountType) && schema) {
+          normalizedQuotes.push({
+            quote: {
+              paymentMethod: PaymentMethod.Bank,
+              fee: value?.fee !== undefined ? parseFloat(value.fee) : null,
+              kycInfo: quote.kyc.kycRequired ? strings.idRequired : null,
+              timeEstimation: getSettlementEstimation(
+                value?.settlementTimeLowerBound,
+                value?.settlementTimeUpperBound
+              ),
+              onPress: () => {
+                ValoraAnalytics.track(FiatExchangeEvents.cico_providers_quote_selected, {
+                  flow,
+                  paymentMethod: PaymentMethod.Bank,
+                  provider: quote.provider.id,
+                })
+                navigate(Screens.FiatDetailsScreen, {
+                  quote,
+                  fiatAccountType: key as FiatAccountType,
+                  fiatAccountSchema: schema.fiatAccountSchema,
+                  allowedValues: schema.allowedValues,
+                  flow,
+                })
+              },
+            },
+            provider: {
+              id: quote.provider.id,
+              name: quote.provider.providerName,
+              logo: quote.provider.imageUrl,
+            },
+          })
+        }
+      }
+    )
+  })
+  return normalizedQuotes
+}
+
+export function normalizeExternalProviders(
+  flow: CICOFlow,
+  input: FetchProvidersOutput[] | undefined
+): NormalizedQuote[] {
+  const normalizedQuotes: NormalizedQuote[] = []
+
+  input?.forEach((provider) => {
+    if (
+      !provider.quote ||
+      provider.restricted ||
+      provider.unavailable ||
+      (flow === CICOFlow.CashIn && !provider.cashIn) ||
+      (flow === CICOFlow.CashOut && !provider.cashOut)
+    )
+      return
```

Comment:
nit: move this to a filter function? E.g., `input?.filter(...conditions).forEach(...)`